src/main/java/it/gov/pagopa/admissibility/service/onboarding/AdmissibilityEvaluatorMediatorServiceImpl.java

```diff
@@ -122,15 +130,39 @@ private EvaluationDTO evaluateOnboardingChecks(OnboardingDTO onboardingRequest,
         } else return null;
     }
 
-    private Mono<EvaluationDTO> retrieveAuthoritiesDataAndEvaluateRequest(OnboardingDTO onboardingRequest, Map<String, Object> onboardingContext, Message<String> message) {
+    private Mono<ImmutablePair<EvaluationDTO, InitiativeConfig>> retrieveAuthoritiesDataAndEvaluateRequest(OnboardingDTO onboardingRequest, Map<String, Object> onboardingContext, Message<String> message) {
         final InitiativeConfig initiativeConfig = readInitiativeConfigFromContext(onboardingContext);
 
         return authoritiesDataRetrieverService.retrieve(onboardingRequest, initiativeConfig, message)
-                .flatMap(r -> onboardingRequestEvaluatorService.evaluate(r, initiativeConfig));
+                .flatMap(r -> onboardingRequestEvaluatorService.evaluate(r, initiativeConfig))
+                .flatMap(n -> Mono.just(ImmutablePair.of(n, initiativeConfig)));
```

Comment:
use map operator instead
flatMap should be used only if reactive or blocking logic is used